src/request.ts

```diff
@@ -224,7 +227,7 @@ export class OData {
           this.credential.username,
           this.credential.password
         ), 
-        cookie: this.sessionCookies
+        cookie: document.cookie
```

Comment:
This statement is wrapped by `if(this.credential)`, and only will return headers with cookie if client manually set user/password credential. But in fact almost all SAP clients are protected by SAML SSO, and client do not need to input the user/password.

If you want to use cookie value in newBatchRequest func, just move it to `var rt = { ...this.commonHeader }`;
If not, just remove it.

Thanks.

Reply:
About the `document.cookie`, it only works if the code is running in a browser, but since node is aimed to backend this won't be available. 

Any suggestions?

Reply:
Okay, three comments:

* Do you want to use this lib in backend (for nodejs ?), or use this lib in frontend (in react/vuejs/ui5)
* For browser, if the S/4HANA OData implementation required cookie and csrf token in `batch sub http request payload`, just add this logic into `c4codata` code. in another words, generally odata `batch` request only require `cookie` and `token` in the HTTP header (instead of `sub requests http header` in payload)
* For nodejs, if you want to keep cookie and session like browser, you need some HTML5 `fetch` polyfill tool. e.g.

```js
const { OData } = require("c4codata")

global.fetch = require('fetch-cookie')(require('node-fetch'))
// you could use `fetch` in nodejs and cookie will be auto processed now
```

Reply:
Should be also possible to use it on the backend. But I'll retract the changes on the cookie and check if this will still be needed, if so I'll create a new PR